### PR TITLE
MODINVSTOR-322 Inventory App: Assign tags to item record

### PR DESCRIPTION
**Pull request by stanislav-mureiko**
## Purpose
Due to https://issues.folio.org/browse/MODINVSTOR-322 we would like to introduce tags for item entities.

## Approach
* updated item model with tags
* updated item interfaces versions

### CODE REVIEW

**Comment by stanislav-mureiko** [comment]
This PR should be merged after https://github.com/folio-org/mod-inventory#302

File: src/test/java/org/folio/rest/api/ItemStorageTest.java [hunk 1]
```diff
@@ -777,6 +782,47 @@ public void canSearchForItemsByBarcode()
...
+    assertThat(searchResponse.getStatusCode(), is(200));
+
+    JsonObject searchBody = searchResponse.getJson();
+
+    JsonArray foundItems = searchBody.getJsonArray("items");
+
+    LinkedHashMap item = (LinkedHashMap) foundItems.getList().get(0);
+    LinkedHashMap<String, ArrayList<String>> itemTags = (LinkedHashMap<String, ArrayList<String>>) item.get("tags");
+
+    assertTrue(searchResponse.getBody().contains(tagValue));
```
Comment (kostyantyn-kh) [inline src/test/java/org/folio/rest/api/ItemStorageTest.java]: Could you move asserts for _foundItems_, _searchResponse_ and _searchBody_ under item? to check, for example, items size before you retrieve first element from list

File: src/test/java/org/folio/rest/api/ItemStorageTest.java [hunk 2]
```diff
@@ -777,6 +782,47 @@ public void canSearchForItemsByBarcode()
       is("673274826203"));
   }
 
+  @Test
+  public void canSearchForItemsByTags()
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/api/ItemStorageTest.java]: Personally, I would also update the tests for creating and updating items to include tags as well.

(I think that is also relevant for the previous changes for instances, apologies that I didn't manage to provide that feedback during those pull requests)

File: src/test/java/org/folio/rest/api/ItemStorageTest.java (hunk 1 above)
Comment (stanislav-mureiko) [reply src/test/java/org/folio/rest/api/ItemStorageTest.java]: Thanks, I'll fix it.

File: src/test/java/org/folio/rest/api/ItemStorageTest.java (hunk 2 above)
Comment (stanislav-mureiko) [reply src/test/java/org/folio/rest/api/ItemStorageTest.java]: Thanks it really makes sense.

**Pull request merged**